### PR TITLE
refactor: factor out maputils

### PR DESCRIPTION
**Pull request by acpana**
Proof of concept of sharing the map utils and other utils across all the `direct` package.

### CODE REVIEW

**Comment by acpana** [comment]
/hold

A few more places that can be factored out if this is something we want to do!

**Comment by yuwenma** [comment]
Actually I think we shall move the cbwp mapping to the direct controller dir (but since we are still developing the mapper tool, this may not be done anyways 😆 ).

**Comment by acpana** [comment]
I don't mind taking on the tooling changes for this too.

File: pkg/controller/direct/maputils.go
Comment (acpana) [inline pkg/controller/direct/maputils.go:1]: Also I think it makes sense to do this now as we are starting to add more resources as opposed to doing it later.

**Review by justinsb — APPROVED** [review]

**Pull request merged**